error whiling building CppMicroServices/doc/src/examples/getting_started/

$cd CppMicroServices/doc/src/examples/getting_started/
$mkdir build && cd $_
$cmake ..
$make

➜  build git:(development) ✗ sudo make
[  5%] Building CXX object CMakeFiles/ServiceTime.dir/service_time/ServiceTime.cpp.o
[ 10%] Linking CXX static library libServiceTime.a
[ 10%] Built target ServiceTime
[ 15%] Checking resource dependencies for ServiceTime_SystemClock
[ 21%] Checking resource dependencies for ServiceTime_SystemClock
[ 26%] Building CXX object CMakeFiles/ServiceTime_SystemClock.dir/service_time_systemclock/ServiceTimeImpl.cpp.o
[ 31%] Building CXX object CMakeFiles/ServiceTime_SystemClock.dir/ServiceTime_SystemClock/cppmicroservices_resources.cpp.o
[ 36%] Building CXX object CMakeFiles/ServiceTime_SystemClock.dir/ServiceTime_SystemClock/cppmicroservices_init.cpp.o
[ 42%] Linking CXX static library libServiceTime_SystemClock.a
Appending zipped resources to ServiceTime_SystemClock
[ 42%] Built target ServiceTime_SystemClock
[ 47%] Checking resource dependencies for ServiceTime_Consumer
[ 52%] Checking resource dependencies for ServiceTime_Consumer
[ 57%] Building CXX object CMakeFiles/ServiceTime_Consumer.dir/service_time_consumer/ServiceTimeConsumer.cpp.o
[ 63%] Building CXX object CMakeFiles/ServiceTime_Consumer.dir/ServiceTime_Consumer/cppmicroservices_resources.cpp.o
[ 68%] Building CXX object CMakeFiles/ServiceTime_Consumer.dir/ServiceTime_Consumer/cppmicroservices_init.cpp.o
[ 73%] Linking CXX static library libServiceTime_Consumer.a
Appending zipped resources to ServiceTime_Consumer
[ 73%] Built target ServiceTime_Consumer
[ 78%] Checking resource dependencies for GettingStarted
Warning: --bundle-name option is unnecessary here.
[ 84%] Checking resource dependencies for GettingStarted
[ 89%] Building CXX object CMakeFiles/GettingStarted.dir/main.cpp.o
[ 94%] Building CXX object CMakeFiles/GettingStarted.dir/GettingStarted/cppmicroservices_resources.cpp.o
[100%] Linking CXX executable GettingStarted
/usr/bin/ld: cannot open output file GettingStarted: Is a directory
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/GettingStarted.dir/build.make:129: GettingStarted] Error 1
make[1]: *** [CMakeFiles/Makefile2:169: CMakeFiles/GettingStarted.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

--- a/doc/src/examples/getting_started/CMakeLists.txt
+++ b/doc/src/examples/getting_started/CMakeLists.txt
@@ -1,3 +1,5 @@
+cmake_minimum_required(VERSION 3.17)
+
 # [proj-begin]
 project(CppMicroServicesGettingStarted)
 
@@ -5,6 +7,23 @@ find_package(CppMicroServices REQUIRED)
 # [proj-end]
 
 include(GenerateExportHeader)
+
+# Detect "local" build (build from getting started directory)
+if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  set(IS_LOCAL_BUILD ON)
+endif()
+
+if (IS_LOCAL_BUILD)
+  # Set binary and library output directories
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+
+  set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries")
+  set(US_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+
+  set(CMAKE_ENABLE_EXPORTS ON)
+endif()
 
 # [interface-begin]
 #=========================================================
@@ -127,8 +146,8 @@ if(NOT BUILD_SHARED_LIBS)
   usFunctionGetResourceSource(TARGET GettingStarted OUT _srcs)
 endif()
 
-add_executable(GettingStarted ${_srcs})
-
+add_executable(GettingStarted)
+target_sources(GettingStarted PRIVATE ${_srcs})
 target_link_libraries(GettingStarted CppMicroServices)
 
 if(NOT BUILD_SHARED_LIBS)

--- a/doc/src/examples/getting_started/CMakeLists.txt
+++ b/doc/src/examples/getting_started/CMakeLists.txt
@@ -10,10 +10,6 @@ include(GenerateExportHeader)
 
 # Detect "local" build (build from getting started directory)
 if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
-  set(IS_LOCAL_BUILD ON)
-endif()
-
-if (IS_LOCAL_BUILD)
   # Set binary and library output directories
   set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")

